### PR TITLE
Handle null message field in protocol exception factory

### PR DESCRIPTION
**Pull request by ihsandemir**
Fix for handling exception creation when the message is null. Also, changed the remote controller version from 0.4 to 0.3.

### CODE REVIEW

File: scripts/start-rc.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ HZ_VERSION="3.11"
 HAZELCAST_TEST_VERSION=${HZ_VERSION}
 HAZELCAST_VERSION=${HZ_VERSION}
 HAZELCAST_ENTERPRISE_VERSION=${HZ_VERSION}
-HAZELCAST_RC_VERSION="0.4-SNAPSHOT"
+HAZELCAST_RC_VERSION="0.3-SNAPSHOT"
```
Comment (SaitTalhaNisanci) [inline scripts/start-rc.sh]: what is the reason for this change?

Comment (ihsandemir) [reply scripts/start-rc.sh]: Because the compatibility tests are using this lower version and I wanted both to be the same version. I
